homeassistant/components/evse_wifi/config_flow.py

```diff
@@ -0,0 +1,101 @@
+"""Config flow for EVSE Wifi integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import (
+    CONF_HOST,
+    CONF_INTERVAL,
+    CONF_MAX_CURRENT,
+    CONF_NAME,
+    DEFAULT_NAME,
+    DOMAIN,
+)
+from .evse import EVSE
+
+_LOGGER = logging.getLogger(__name__)
+
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default=DEFAULT_NAME): str,
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_MAX_CURRENT, default=32): int,
+        vol.Required(CONF_INTERVAL, default=15): int,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+
+    evse = EVSE(
+        host=str(data.get(CONF_HOST)),
+        name=str(data.get(CONF_NAME)),
+        max_current=int(data.get(CONF_MAX_CURRENT, 32)),
+        interval=int(data.get(CONF_INTERVAL, 15)),
+    )
+
+    test_result = await evse.test()
+    if test_result is False:
+        raise CannotConnect
+
+    # If you cannot connect:
+    # throw CannotConnect
+    # If the authentication is wrong:
+    # InvalidAuth
+
+    # Return info that you want to store in the config entry.
```

Comment:
Remove the comments.

Check authentication and cannot connect separately ideally.